### PR TITLE
Update Dashboard confirm logic

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
- Add assert if Dashboard uses non-deferred intent when PaymentSheet loads. Non-deferred intent flows aren't supported today because they don't create a PM first, which Dashboard's `uk_` key requires.
- Refactor out unnecessary dashboard helper methods
- Add support for Dashboard SetupIntents: Set moto and make API accommodations for `uk_` to retrieve and confirm SetupIntent API calls, just like https://github.com/stripe-ios/stripe-ios-private/pull/59 

## Motivation
https://jira.corp.stripe.com/browse/RUN_MOBILESDK-2808

## Testing
Added tests (h/t @mludowise-stripe) for Dashboard PI and SI confirm flows that
- Use Dashboard's "id" shaped client secrets
- Check moto is set
- Check client secret is nil (for whatever reason, this is a requirement for uk)

I also manually tested Dashboard's deferred payment and setup flow. 
* PaymentIntent w/ new card: https://admin.corp.stripe.com/payment_intent/pi_3OUc1uL6pqDH2fDJ1DYclWI2
* PaymentIntent w/ saved card: https://admin.corp.stripe.com/payment_intent/pi_3OUcNML6pqDH2fDJ1YQEiKXl
* SetupIntent (using this [branch](https://github.com/stripe-ios/dashboard-ios/tree/stakr/nschris/navIAM1-17)) : https://admin.corp.stripe.com/setup_intent/seti_1OUcMfL6pqDH2fDJ7ATF3bD1

## Changelog
Not user facing

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift [hunk 1]
```diff
@@ -33,6 +33,11 @@ final class PaymentSheetLoader {
 
         Task { @MainActor in
             do {
+                if !mode.isDeferred && configuration.apiClient.publishableKeyIsUserKey {
+                    // User keys can't pass payment_method_data directly to /confirm, which is what the non-deferred intent flows do
+                    assertionFailure("Dashboard isn't supported in non-deferred intent flows")
```
Comment (mludowise-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift:38]: ➕

**Review by mludowise-stripe — DISMISSED** [review]
Looks great! I'm happy we're migrating the logic into `makePaymentIntentParams` so it's more centralized and the test coverage is much more robust.

Thanks for taking this.

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheet+DashboardConfirmParamsTest.swift [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        let exp = expectation(description: "confirm payment requested")
+
+        stub { urlRequest in
+            guard let pathComponents = urlRequest.url?.pathComponents else { return false }
+            return pathComponents.last == "confirm"
+        } response: { [self] request in
+            XCTAssertFalse(request.url!.absoluteString.contains("secret")) // Sanity check that the request /confirm URL doesn't contain a client secret
+            let params = bodyParams(from: request, line: line)
+
+            // For unknown reasons, the API doesn't allow Dashboard `uk_` keys to pass payment_method_data
```
[the remark this answers is not in the document]
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheet+DashboardConfirmParamsTest.swift:277]: Oh nice, you found the source!

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheet+DashboardConfirmParamsTest.swift [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+            authenticationContext: self,
+            intent: .deferredIntent(elementsSession: .emptyElementsSession, intentConfig: MockParams.deferredPaymentIntentConfiguration(clientSecret: MockParams.dashboardPaymentIntentClientSecret)),
+            paymentOption: .saved(paymentMethod: MockParams.cardPaymentMethod),
+            paymentHandler: STPPaymentHandler(apiClient: configuration.apiClient),
+            completion: { _, _ in
+                exp.fulfill()
+            }
+        )
+
+        waitForExpectations(timeout: 10)
```
[the remark this answers is not in the document]
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheet+DashboardConfirmParamsTest.swift:151]: 10s is our typical timeout, I think it's okay.

**Review by porter-stripe — APPROVED** [review]

**Review by mludowise-stripe — APPROVED** [review]

**Pull request merged**